lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -102,22 +102,24 @@ def server_version
       kubectl.server_version
     end
 
-    def initialize(namespace:, context:, current_sha:, template_dir:, logger:, kubectl_instance: nil, bindings: {},
+    def initialize(namespace:, context:, current_sha:, template_dirs:, logger:, kubectl_instance: nil, bindings: {},
       max_watch_seconds: nil, selector: nil)
       @namespace = namespace
       @namespace_tags = []
       @context = context
       @current_sha = current_sha
-      @template_dir = File.expand_path(template_dir)
+      @template_dirs = template_dirs.map { |template_dir| File.expand_path(template_dir) }
       @logger = logger
       @kubectl = kubectl_instance
       @max_watch_seconds = max_watch_seconds
-      @renderer = KubernetesDeploy::Renderer.new(
-        current_sha: @current_sha,
-        template_dir: @template_dir,
-        logger: @logger,
-        bindings: bindings,
-      )
+      @renderers = @template_dirs.map do |template_dir|
```

Comment:
It might be worth switching to a hash with a default block: 
``` ruby
@renderers = Hash.new do |hash, template_dir|
  hash[template_dir] = KubernetesDeploy::Renderer.new(
    current_sha: @current_sha,
    logger: @logger,
    template_dir: template_dir,
    bindings: bindings,
  )
end
```

Accessing it with `@renderers[template_dir]` further down would instantiate it the first time and return the first one from then on.

Reply:
This turned out to be a really good idea, since the closure exposes `current_sha, logger, and bindings` so they don't have to be passed around elsewhere